randovania/games/samus_returns/exporter/hint_namer.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import TYPE_CHECKING
+
+from randovania.exporter.hints.hint_formatters import LocationFormatter, RelativeAreaFormatter, TemplatedFormatter
+from randovania.exporter.hints.hint_namer import HintNamer, PickupLocation
+from randovania.exporter.hints.relative_item_formatter import RelativeItemFormatter
+from randovania.game_description import default_database
+from randovania.game_description.hint import Hint, HintLocationPrecision
+
+if TYPE_CHECKING:
+    from randovania.exporter.hints.pickup_hint import PickupHint
+    from randovania.game_description.game_patches import GamePatches
+    from randovania.games.game import RandovaniaGame
+    from randovania.interface_common.players_configuration import PlayersConfiguration
+
+
+class MSRHintNamer(HintNamer):
+    location_formatters: dict[HintLocationPrecision, LocationFormatter]
+
+    def __init__(self, all_patches: dict[int, GamePatches], players_config: PlayersConfiguration):
+        patches = all_patches[players_config.player_index]
+        location_hint_template = "{determiner.title}{pickup} can be found in {node}."
+
+        self.location_formatters = {
+            HintLocationPrecision.DETAILED: TemplatedFormatter(
+                location_hint_template,
+                self,
+            ),
+            HintLocationPrecision.REGION_ONLY: TemplatedFormatter(
+                location_hint_template,
+                self,
+            ),
+            HintLocationPrecision.RELATIVE_TO_AREA: RelativeAreaFormatter(
+                patches,
+                lambda msg, with_color: msg,
+            ),
+            HintLocationPrecision.RELATIVE_TO_INDEX: RelativeItemFormatter(
+                patches,
+                lambda msg, with_color: msg,
+                players_config,
+            ),
+        }
+
+    def format_region(self, location: PickupLocation, with_color: bool) -> str:
+        region_list = default_database.game_description_for(location.game).region_list
+        result = region_list.region_name_from_node(region_list.node_from_pickup_index(location.location), True)
+        return result
+
+    def format_location_hint(self, game: RandovaniaGame, pick_hint: PickupHint, hint: Hint, with_color: bool) -> str:
+        assert hint.precision is not None
+        msg = self.location_formatters[hint.precision.location].format(
+            game,
+            dataclasses.replace(pick_hint, pickup_name=pick_hint.pickup_name),
+            hint,
+            with_color,
+        )
+        return f"{msg}\n"
+
+    def format_temple_name(self, temple_name: str, with_color: bool) -> str:
+        raise RuntimeError("Unsupported feature")
```

Comment:
Since color isn't a thing, shouldn't this be removed?

Reply:
In general, yes, but in this case it's implementing the methods from the base class: https://github.com/randovania/randovania/blob/main/randovania/exporter/hints/hint_namer.py